usr/lib/ica_s390_stdll/ica_specific.c

```diff
@@ -6661,3 +6661,25 @@ CK_RV token_specific_object_add(STDLL_TokData_t *tokdata, SESSION *sess,
         return CKR_OK;
     }
 }
+
+CK_RV token_specific_get_token_info(STDLL_TokData_t *tokdata,
+                                    CK_TOKEN_INFO_PTR pInfo)
+{
+    libica_version_info ver;
+    int rc;
+
+    UNUSED(tokdata);
+
+    rc = ica_get_version(&ver);
```

Comment:
Hmm, just a thought ... you could obtain the libica fips status via ica_fips_status() and put this info into the token info as well ...